### ISSUE TITLE
torch.matrix_exp gives incorrect results for complex tensors

### ISSUE THREAD

**Issue by auroraustc** [pytorch/pytorch#48299]
## 🐛 Bug

torch.matrix_exp gives incorrect results if the elements of the tensor are complex numbers.

## To Reproduce

```bash
>>> import numpy as np
>>> import torch
>>> import scipy.linalg
>>> a = np.array([[1.0, 2.0], [-1.0, 3.0]])
>>> a_torch = torch.tensor(a)
>>> scipy.linalg.expm(a)
array([[-2.22535226, 12.43535262],
       [-6.21767631, 10.21000036]])
>>> torch.matrix_exp(a_torch)
tensor([[-2.2254, 12.4354],
        [-6.2177, 10.2100]], dtype=torch.float64)
>>> scipy.linalg.expm(1j*a)
array([[ 0.4264593 +1.89217551j, -2.13721484-0.97811252j],
       [ 1.06860742+0.48905626j, -1.71075555+0.91406299j]])
>>> torch.matrix_exp(1j*a_torch)
tensor([[ 2.1136e+08-32631878.1690j, -1.6073e+07-62782284.5732j],
        [ 8.0363e+06+31391142.2866j,  1.9529e+08-95414162.7421j]],
       dtype=torch.complex128)
```

## Expected behavior

torch.matrix_exp should give the same result as scipy.linalg.expm

## Environment

Collecting environment information...
PyTorch version: 1.8.0.dev20201119
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.10.2

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX TITAN
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.4
[pip3] torch==1.8.0.dev20201119
[pip3] torchaudio==0.8.0.dev20201108
[pip3] torchvision==0.9.0.dev20201119
[conda] torch                     1.8.0.dev20201119          pypi_0    pypi
[conda] torchaudio                0.8.0.dev20201108          pypi_0    pypi
[conda] torchvision               0.9.0.dev20201119          pypi_0    pypi

## Additional context

cc @ezyang @gchanan @zou3519 @bdhirsh @anjali411 @dylanbespalko @mruberry @rgommers @heitorschueroff

**Comment by gchanan** [comment]
`torch.matrix_exp` was added in 1.7.0.

**Comment by nikitaved** [comment]
My call. Matrix exp was implement before the complex matrix product, hence was not properly tested.

**Comment by gchanan** [comment]
@nikitaved  we are releasing 1.7.1 soon -- should we disable complex matrix_exp?

**Comment by nikitaved** [comment]
How much time do I have? It has been a while since I touched the code, but the output looks like there is some memory access issues. I wrote a custom kernel for handling small linear combinations of tensors, the issue is likely there I guess.

**Comment by nikitaved** [comment]
I think I know where the bug is, will test is soon.

**Comment by nikitaved** [comment]
@gchanan , I have submitted the fix here : https://github.com/pytorch/pytorch/pull/48363 . I think it could be shipped with 1.7.1.
